### PR TITLE
Post images to be cards

### PR DESCRIPTION
**Pull request by mixonic**
TODO:

* Add a default image card implementation to mobiledoc DOM and HTML renderers

### CODE REVIEW

File: demo/demo.js [hunk 1]
```diff
@@ -182,7 +182,8 @@ var ContentKitDemo = exports.ContentKitDemo = {
       'simple-card': simpleCard,
       'edit-card': cardWithEditMode,
       'input-card': cardWithInput,
-      'selfie-card': selfieCard
+      'selfie-card': selfieCard,
+      'image': ContentKit.ImageCard
```
Comment (mixonic) [inline demo/demo.js]: need to add this to the list if not present (in the editor constructor)

Comment (mixonic) [inline demo/demo.js]: Not in the editor constructor, but should be added to the default renderer (some version should be there)

File: src/js/commands/image.js [hunk 2]
```diff
@@ -1,74 +1,24 @@
...
-          this.editor.trigger('update');
-        } else {
-          this.editor.removeSection(imageSection);
-          new Message().showError(error.message || 'Error uploading image');
-        }
-        this.editor.rerender();
-      }
-    });
+    let {post} = this.editor;
+    let sections = this.editor.cursor.activeSections;
```
Comment (bantic) [inline src/js/commands/image.js]: should just be `this.editor.activeSections`

File: src/js/commands/image.js [hunk 3]
```diff
@@ -1,74 +1,24 @@
...
+    let lastSection = sections[sections.length - 1];
+    let section = this.builder.generateCardSection('image');
+    post.insertSectionAfter(section, lastSection);
+    sections.forEach(section => section.renderNode.scheduleForRemoval());
```
Comment (bantic) [inline src/js/commands/image.js]: why does it remove all the sections before this card?

File: tests/acceptance/embed-intent-test.js [hunk 4]
```diff
@@ -43,3 +61,25 @@ Helpers.skipInPhantom('typing inserts section', (assert) => {
...
+  Helpers.dom.triggerEvent(editorElement.childNodes[1].firstChild, 'click');
+
+  let done = assert.async();
+  setTimeout(() => { // delay due to internal async
+    assert.ok($('.ck-embed-intent').is(':visible'), 'embed intent appears');
+
+    Helpers.dom.triggerEvent($('.ck-embed-intent-btn')[0], 'click');
+    Helpers.dom.triggerEvent($('button[title=image]')[0], 'click');
+
+    assert.ok($('.ck-embed-intent').length === 0, 'embed intent is hidden');
```
Comment (bantic) [inline tests/acceptance/embed-intent-test.js]: there is `assert.hasElement` and `assert.hasNoElement`

**Comment by bantic** [comment]
:+1: looks good to me. I had a few questions but go ahead and merge at will

File: src/js/commands/image.js (hunk 3 above)
Comment (mixonic) [inline src/js/commands/image.js]: the selected sections, or just the section the cursor is on, should be replaced. The common scenario is that the intent tools are showing on a blank line and that blank line is replaced with the image.

**Comment by mixonic** [comment]
Upstream mobiledoc renderers now have support for default image cards.

File: demo/demo.js (hunk 1 above)
Comment (mixonic) [inline demo/demo.js]: upstream lib now has a default.

**Pull request merged**